service-front/module/Application/src/Helpers/LocalisationHelper.php

```diff
@@ -0,0 +1,73 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Application\Helpers;
+
+use Application\Exceptions\LocalisationException;
+
+class LocalisationHelper
+{
+    /**
+     * @throws LocalisationException
+     */
+    public function getInternationalSupportedDocuments(array $detailsData): array
+    {
+        $config = $this->getConfig();
+
+        $idDocuments = $config['opg_settings']['supported_countries_documents'];
+        $documents = [];
+
+        if (! isset($detailsData['idMethodIncludingNation']['country'])) {
+            throw new LocalisationException("Country for document list has not been set.");
+        }
+
+        foreach ($idDocuments as $countryDocumentBody) {
+            if ($countryDocumentBody['code'] == $detailsData['idMethodIncludingNation']['country']) {
+                $documents = $countryDocumentBody;
+            }
+        }
+
+        if ($documents === []) {
+            throw new LocalisationException(
+                "Country for document list has not been found in config."
+            );
+        }
+        return $this->processDocumentBody($documents);
+    }
+
+    public function processDocumentBody(array $documents): array
+    {
+        foreach ($documents['supported_documents'] as $key => $value) {
+            $string = $this->parseWord($value['type']);
+
+            $documents['supported_documents'][$key] = array_merge(
+                $documents['supported_documents'][$key],
+                ['display_text' => $string]
+            );
+        }
+        return $documents;
+    }
+
+    public function parseWord(string $word): string
```

Comment:
It would be much more flexible to just have a mapping from type of document to display name:
```php
$display = [
  'passport' => 'Passport',
  ...
];
```
You could make it [a filter](https://twig.symfony.com/doc/3.x/advanced.html#filters) in TwigExtension, rather than in the controller.